packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/write_batch.dart

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of cloud_firestore_platform_interface;
+
+/// A [WriteBatch] is a series of write operations to be performed as one unit.
+///
+/// Operations done on a [WriteBatch] do not take effect until you [commit].
+///
+/// Once committed, no further operations can be performed on the [WriteBatch],
+/// nor can it be committed again.
+class WriteBatch extends WriteBatchPlatform {
+  WriteBatch(this._firestore)
+      : _handle = MethodChannelFirestore.channel.invokeMethod<dynamic>(
+            'WriteBatch#create',
+            <String, dynamic>{'app': _firestore.appName()}),
+        super._();
+
+  final FirestorePlatform _firestore;
+  Future<dynamic> _handle;
+  final List<Future<dynamic>> _actions = <Future<dynamic>>[];
+
+  @override
+  Future<void> commit() async {
+    if (!_committed) {
```

Comment:
You can extract all this `if (!_commited) { blah } else { throw }` to a function:

```dart
void _assertNotCommitted() {
  if (_committed) {
    throw StateError('This batch has already been committed and can no longer be changed.');
  }
}
```

Then you can call this at the beginning of each method that is guarded like this, so you don't need the if/else check everywhere.